File: MATLAB/EncryptVigenere_German.m
function y = EncryptVigenere_German(x, k)
k1 = double(k) - 97;
n = size(x, 2);
m = size(k, 2);
for i = 1 : n 
    xp = double(x(i)) - 97;
    ii = mod(i - 1, m) + 1;
    y(i) = mod(xp + k1(ii), 30) + 65;
end;
y = char(y);

File: Julia/EncryptVigenere_German.jl
function EncryptVigenere_German(plaintext_ascii_30, keyword)
    keyword_ascii = [Int(i) - 97 for i in keyword]
    keyword_ascii = repmat(keyword_ascii, cld(length(plaintext_ascii_30), length(keyword_ascii)))
    pop_length = length(keyword_ascii) - length(plaintext_ascii_30) - 1
    keyword_ascii = deleteat!(keyword_ascii, length(keyword_ascii) - pop_length:length(keyword_ascii))
    return mod.((plaintext_ascii_30 - 97) + keyword_ascii, 30) + 65
end
